applications/machine_learning/src/modules/ei_data_forwarder_bt_nus.c

```diff
@@ -39,19 +43,44 @@ enum state {
 	STATE_ERROR
 };
 
+struct ei_data_packet {
+	sys_snode_t node;
+	size_t size;
+	uint8_t buf[DATA_BUF_SIZE];
+};
+
 static uint8_t conn_state;
 static struct bt_conn *nus_conn;
 
-static atomic_t nus_busy;
 static enum state state = ML_STATE_CONTROL ? STATE_DISABLED_SUSPENDED : STATE_DISABLED_ACTIVE;
 
+BUILD_ASSERT(sizeof(struct ei_data_packet) % BUF_SLAB_ALIGN == 0);
+K_MEM_SLAB_DEFINE(buf_slab, sizeof(struct ei_data_packet), DATA_BUF_COUNT, BUF_SLAB_ALIGN);
+
+static sys_slist_t send_queue;
+static struct k_work send_queued;
+static size_t pipeline_cnt;
+static atomic_t sent_cnt;
+
 
 static void report_error(void)
 {
 	state = STATE_ERROR;
 	module_set_state(MODULE_STATE_ERROR);
 }
 
+static void clean_buffered_data(void)
+{
+	sys_snode_t *node = sys_slist_get(&send_queue);
+
+	while (node) {
```

Comment:
why not `while (node = sys_slist_get())`

Reply:
Changed